component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/smsotp/SMSOTPAuthenticator.java

```diff
@@ -25,6 +25,7 @@
 import org.apache.commons.lang.math.NumberUtils;
 import org.apache.commons.logging.Log;
 import org.apache.commons.logging.LogFactory;
+import org.apache.logging.log4j.ThreadContext;
```

Comment:
shall we check the possbility if using slf4j as mentioned in [1]
[1] https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/1822

Reply:
noted on the above. I will check this.

Reply:
as log4j2 is also used [1] apart from slf4j, we can make use of the threadcontext here. 
[1] https://github.com/wso2/identity-api-user/pull/142

Reply:
Hi @Tiffany-silva ,
When adding new logs, it is recommended to use the [SLF4J abstract logging API](https://www.javadoc.io/doc/org.slf4j/slf4j-api/latest/org.slf4j/org/slf4j/package-summary.html)  instead of the log4j 2.x library to guarantee a decoupling between the source and the logging framework at deployment. This will give us flexibility in changing the logging implementation later if needed.

Reply:
Thanks for the clarification. let's use the recommended library.